File: Engine/src/jvmMain/java/org/etieskrill/engine/input/CursorInputHandler.java
package org.etieskrill.engine.input;

import org.jetbrains.annotations.NotNull;

public interface CursorInputHandler {

    boolean invokeClick(@NotNull Key button, @NotNull Keys.Action action, double posX, double posY);

    boolean invokeMove(double posX, double posY);

    boolean invokeScroll(double deltaX, double deltaY);

}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/input/CursorInputHandler.kt
package org.etieskrill.engine.input

interface CursorInputHandler {

    fun invokeClick(button: KeyEvent, posX: Double, posY: Double): Boolean

    fun invokeMove(posX: Double, posY: Double): Boolean

    fun invokeScroll(deltaX: Double, deltaY: Double): Boolean

}
